controllers/idler/idler_controller.go

```diff
@@ -144,43 +156,12 @@ func (r *Reconciler) ensureIdling(ctx context.Context, idler *toolchainv1alpha1.
 			// Already tracking this pod. Check the timeout.
 			if time.Now().After(trackedPod.StartTime.Add(time.Duration(timeoutSeconds) * time.Second)) {
 				podLogger.Info("Pod running for too long. Killing the pod.", "start_time", trackedPod.StartTime.Format("2006-01-02T15:04:05Z"), "timeout_seconds", timeoutSeconds)
-				var podreason string
-				podCondition := pod.Status.Conditions
-				for _, podCond := range podCondition {
-					if podCond.Type == "Ready" {
-						podreason = podCond.Reason
-					}
-				}
-
-				// Check if it belongs to a controller (Deployment, DeploymentConfig, etc) and scale it down to zero.
 				lctx := log.IntoContext(ctx, podLogger)
```

Comment:
you can define the pod context only once - when you defined the pod logger

Reply:
done in [bfc3cda](https://github.com/codeready-toolchain/member-operator/pull/630/commits/bfc3cdab10e2b06e72dbf287053a7f7188fad488)